### PR TITLE
CHAOS-6083 Linux API block fault

### PR DESCRIPTION
**Pull request by SmritiSatya**
Linux API block fault docs and fault diagram

### CODE REVIEW

**Review by brian-f-harness — DISMISSED** [review]

File: docs/chaos-engineering/chaos-faults/linux/linux-api-block.md [hunk 1]
```diff
@@ -0,0 +1,402 @@
...
+
+import Ossupport from './shared/note-supported-os.md'
+import FaultPermissions from './shared/fault-permissions.md'
+
+Linux API block injects API block fault into a Linux machine for a specific duration through path filtering. This results in the API not being able to send responses for the requests it receives.
+
+![Linux API block](./static/images/linux-api-block.png)
+
+## Use cases
+- Validates how well your system can handle disruptions in API services for a specific pod.
```
Comment (neelanjan00) [inline docs/chaos-engineering/chaos-faults/linux/linux-api-block.md:14]:
```suggestion
- Validates how well your system can handle API service disruptions for a Linux server.
```

Comment (neelanjan00) [reply docs/chaos-engineering/chaos-faults/linux/linux-api-block.md:14]: Check other places too for pods

File: docs/chaos-engineering/chaos-faults/linux/linux-api-block.md [hunk 2]
```diff
@@ -0,0 +1,402 @@
...
+#### Ingress
+
+Set this parameter if the HTTPS URL of the target application includes a port whose format is `https://<hostname>:port`. However, if the HTTPS URL is in the format `https://<hostname>` without a port, this setting is not required.
+
+#### Egress
+
+For outbound traffic, set `httpsEnabled` to `true` to enable HTTPS support for external services. This enables using TLS certificates for the proxy within the target application.
+
+- If the HTTP client in the target application is configured to reload certificates with each API call, set `httpsEnabled` to `true`. You won't need to provide `customCertificates` input variable.
+- However, if the root certificate directory and file name differ from `/etc/ssl/certs` and `ca-certificates.crt` respectively, provide the root CA certificate file name using `httpsRootCertFile` input variable.
```
Comment (neelanjan00) [inline docs/chaos-engineering/chaos-faults/linux/linux-api-block.md:255]:
```suggestion
- However, if the root CA certificate file path is not `/etc/ssl/certs/ca-certificates.crt`, provide it using `httpsRootCertFile` input variable.
```

File: docs/chaos-engineering/chaos-faults/linux/linux-api-block.md [hunk 3]
```diff
@@ -0,0 +1,402 @@
...
+    appkind: "deployment"
+  chaosServiceAccount: litmus-admin
+  experiments:
+    - name: linux-api-block
+      spec:
+        components:
+          env:
+            # provide the port of the target service
+            - name: targetServicePort
+              value: "80"
```
Comment (neelanjan00) [inline docs/chaos-engineering/chaos-faults/linux/linux-api-block.md:156]: The data types might be different, please check the reference manifests.
```suggestion
              value: 80
```

File: docs/chaos-engineering/chaos-faults/linux/linux-api-block.md [hunk 4]
```diff
@@ -0,0 +1,402 @@
...
+            - name: pathFilter
+              value: '/status'
+            - name: statusCode
+              value: "404"
```
Comment (neelanjan00) [inline docs/chaos-engineering/chaos-faults/linux/linux-api-block.md:160]:
```suggestion
              value: 404
```

File: docs/chaos-engineering/chaos-faults/linux/linux-api-block.md (hunk 3 above)
Comment (neelanjan00) [reply docs/chaos-engineering/chaos-faults/linux/linux-api-block.md:156]: Check other places as well

**Pull request merged**